internal/pkg/csumio/csumreader_test.go

```diff
@@ -184,3 +184,227 @@ func TestReader_CorrectChecksum_VerificationEnabled(t *testing.T) {
 		})
 	}
 }
+
+// TestReader_ReadFactorCoverage exhaustively validates csumio.Reader.Read.
+// The test matrix uses pairwise coverage across three factors:
+//
+//	Factor A: requested len(p) relative to chunkSize (> chunk / == chunk / < chunk).
+//	Factor B: remaining unread bytes relative to len(p) (> p / == p / < p).
+//	Factor C: inner-buffer state (empty vs. partially consumed from a prior chunk).
```

Comment:
`consume` is a bit ambiguous expression.

```suggestion
//	Factor C: inner-buffer state (empty vs. partially filled by a prior read).
```